pkg/cloudprovider/provider/cherryservers/provider.go

```
@@ -0,0 +1,429 @@
+	}
+	c.Plan, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.Plan)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"plan\" field, error = %v", err)
+	}
+
+	c.ProjectID, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.ProjectID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"projectID\" field, error = %v", err)
+	}
+
+	c.TeamID, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.TeamID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"teamID\" field, error = %v", err)
+	}
+
+	c.Location, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.Location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"location\" field, error = %v", err)
+	}
+	c.Tags = rawConfig.Tags
+	return &c, &pconfig, err
+}
+
+func getNameForOS(os providerconfig.OperatingSystem) (string, error) {
+	switch os {
+	case providerconfig.OperatingSystemUbuntu:
+		return "Ubuntu 18.04 64bit", nil
+	case providerconfig.OperatingSystemCentOS:
+		return "CentOS 7 64bit", nil
+	}
+	return "", providerconfig.ErrOSNotSupported
+}
+
+func getClient(token string) *cherrygo.Client {
+	httpClient := &http.Client{}
+	return cherrygo.NewClientWithAuthVar(httpClient, token)
+}
+
+func getTeamByID(client *cherrygo.Client, ID string) (*cherrygo.Teams, error) {
+	teams, _, err := client.Teams.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, team := range teams {
+		if strconv.Itoa(team.ID) == ID {
+			return &team, nil
+		}
+	}
+	return nil, nil
+}
+
+func getPlanByName(client *cherrygo.Client, planName string, teamID string) (*cherrygo.Plans, error) {
+	tid, err := strconv.Atoi(teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	plans, _, err := client.Plans.List(tid)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, plan := range plans {
+		if plan.Name == planName {
+			return &plan, nil
+		}
+	}
+	return nil, nil
```

Same as for `getTeamByID` - should we return an error here?

---

Fixed. Same as above